### PR TITLE
Wishlist

### PR DESCRIPTION
**Pull request by EKustova**
create Wishilist feature file

### CODE REVIEW

File: ChainReactionTests/Wishlist.feature [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+	And I click Add To List button
+	Then I see "Item added to wishlist" message on wishlist popup
+	And I click Close button
+
+	When I click Wishilist button
+	Then I see list with names of wishilist
+	And I see "1" item in wishlist with name "My Wishlist"
+
+#given account with 2 wishlists
+Scenario: Verify select wishlist to add new good
```
Comment (etolmachev) [inline ChainReactionTests/Wishlist.feature]: Use account without wishlists, add lists during scenario

File: ChainReactionBindings/Bindings/MainPageBidings/MainPageBindings.cs [hunk 2]
```diff
@@ -35,5 +35,11 @@ public void WhenIClickSearchButton()
 		{
 			page.SearchClick();
 		}
+
+		[When(@"I click Wishlist button")]
```
Comment (etolmachev) [inline ChainReactionBindings/Bindings/MainPageBidings/MainPageBindings.cs]: Let's move this step to TopMenu Bindings, also please change to 
"I click Wishlist button on Top Menu"

File: ChainReactionBindings/Bindings/WishlistPageBindings.cs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+			Assert.AreEqual(message, page.EmptyWishlistElement.Text);
+		}
+
+		[Then(@"I see my wishlist with ""(.*)"" items")]
+		public void SeeMyWishlistWithItems(string count)
+		{
+			Assert.AreEqual(count, page.CountItemsElement.Text);
+		}
+
+		[Then(@"I see ""(.*)"" wishlist in list of wishilist")]
```
Comment (etolmachev) [inline ChainReactionBindings/Bindings/WishlistPageBindings.cs]: Let's use "wishlists" to make it more readable

File: ChainReactionBindings/Bindings/WishlistPageBindings.cs [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+			var el = page.FindWishlist(name);
+			var countItems = el.FindElement(By.CssSelector(page.CountItemsId));
+			Assert.AreEqual(count, countItems.Text);
+		}
+
+		[When(@"I click View List button on wishlist with name ""(.*)""")]
+		public void ClickViewListButtonOnWishlistWithName(string name)
+		{
+			var el = page.FindWishlist(name);
+			var viewButton = el.FindElement(By.ClassName("blue_btn2"));
```
Comment (etolmachev) [inline ChainReactionBindings/Bindings/WishlistPageBindings.cs]: Can we hide FindElement and id etc. to the PageObject? Let's create method "GetViewButton"

File: ChainReactionBindings/Bindings/WishlistPageBindings.cs [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+		public void SeeWishlistInListOfWishlist(int count)
+		{
+			Assert.AreEqual(count, page.TableOfWishlistElement.FindElements(By.ClassName("wishlist_col2")).Count);
+		}
+
+		[Then(@"I see ""(.*)"" items in wishlist with name ""(.*)""")]
+		public void SeeItemsInWishlistWithName(string count, string name)
+		{
+			var el = page.FindWishlist(name);
+			var countItems = el.FindElement(By.CssSelector(page.CountItemsId));
```
Comment (etolmachev) [inline ChainReactionBindings/Bindings/WishlistPageBindings.cs]: Can we hide FindElement and id etc. to the PageObject? Let's create method "GetItemsCount"

File: ChainReactionBindings/Bindings/WishlistPageBindings.cs [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+						throw new Exception(string.Format("Field {0} is not implemented", fieldName));
+				}
+			}
+		}
+
+		[When(@"I click Delete button on wishlist with name ""(.*)""")]
+		public void ClickDeleteButtonOnWishlistWithName(string name)
+		{
+			var el = page.FindWishlist(name);
+			var deleteWishlist = el.FindElement(By.ClassName("wishlist_remove"));
```
Comment (etolmachev) [inline ChainReactionBindings/Bindings/WishlistPageBindings.cs]: Same this as for ItemsCount and ViewButton

File: ChainReactionBindings/Bindings/WishlistPageBindings.cs [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+			page.DeleteWishlist(deleteWishlist);
```
Comment (etolmachev) [inline ChainReactionBindings/Bindings/WishlistPageBindings.cs]: We can pass to this function our "el" and hide FindElement inside of PageObject

File: ChainReactionBindings/Bindings/WishlistPopupPageBindings.cs [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+	[Binding]
+	class WishlistPopupPageBindings
+	{
+		WishlistPopupPage page = new WishlistPopupPage();
+
+		[Then(@"I see add to wishlist popup is loaded")]
+		public void ThenISeeAddToWishlistPopupIsLoaded()
+		{
+			page.WishlistPopupPageLoaded();
+			//Assert.IsTrue(page.AddPopupElement.Displayed);
```
Comment (etolmachev) [inline ChainReactionBindings/Bindings/WishlistPopupPageBindings.cs]: Let's Remove this line

File: ChainReactionBindings/Bindings/WishlistPopupPageBindings.cs [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+		public void ChooseFirstGoodAndClickAddToWishlistButton()
+		{
+			page.AddToWishlistInPop();
+		}
+
+		[Then(@"I see select wishlist popup is loaded")]
+		public void ThenISeeSelectWishlistPopupIsLoaded()
+		{
+			page.SelectPopElement.WaitElementAppears(10);
+			Assert.IsTrue(page.SelectPopElement.Displayed);
```
Comment (etolmachev) [inline ChainReactionBindings/Bindings/WishlistPopupPageBindings.cs]: We can delete this line, WaitElementAppears already check for Displayed and will throw exception if it's false

File: ChainReactionBindings/TestBase/Pages/ItemDetailsPage.cs [hunk 10]
```diff
@@ -22,6 +22,8 @@ class ItemDetailsPage : BasePage
 		public string EmptyReviewsId = "#crcPDPBottomTabsReviewsLink";
```
Comment (etolmachev) [inline ChainReactionBindings/TestBase/Pages/ItemDetailsPage.cs]: Please make all the ids private, we should not give access to them outside the PageObject

File: ChainReactionBindings/TestBase/Pages/WishlistPage.cs [hunk 11]
```diff
@@ -0,0 +1,62 @@
+﻿using System.Runtime.InteropServices;
+using OpenQA.Selenium;
+
+namespace ChainReactionBindings.TestBase.Pages
+{
+	class WishlistPage: BasePage
+	{
+		public string WishlistHeaderId = ".user_profile_headding2";
```
Comment (etolmachev) [inline ChainReactionBindings/TestBase/Pages/WishlistPage.cs]: Please make all the ids private, we should not give access to them outside the PageObject

File: ChainReactionBindings/TestBase/Pages/WishlistPopupPage.cs [hunk 12]
```diff
@@ -0,0 +1,66 @@
+﻿using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+
+namespace ChainReactionBindings.TestBase.Pages
+{
+	class WishlistPopupPage : BasePage
+	{
+		public string AddPopupId = ".addto_wishlist_pop.addcart";
```
Comment (etolmachev) [inline ChainReactionBindings/TestBase/Pages/WishlistPopupPage.cs]: Please make all the ids private, we should not give access to them outside the PageObject

**Review by etolmachev — CHANGES_REQUESTED** [review]

File: ChainReactionBindings/Bindings/WishlistPageBindings.cs (hunk 7 above)
Comment (EKustova) [reply ChainReactionBindings/Bindings/WishlistPageBindings.cs]: done

**Pull request merged**